SocietalConstructionToolTests/RuntimeTests.cs

```diff
@@ -40,7 +42,37 @@ public void TestRuntimeExits()
             // Assert
             contextMock.Verify(c => c.ShouldExit, Times.Exactly(2));
             contextMock.Verify(c => c.GetNextContext(), Times.Once);
-            contextMock.Verify(c => c.AgentHandler.Agents, Times.Once);
+            contextMock.Verify(c => c.AgentHandler.Agents, Times.Exactly(2));
+            contextMock.Verify(c => c.OnTick(), Times.Exactly(2));
+            contextMock.Verify(c => c.OnExit(), Times.Once);
+            contextMock.VerifyNoOtherCalls();
+        }
+
+        [TestMethod]
+        public void TestRuntimeExitsWhenNoAgents()
+        {
+            // Arrange
+            var contextMock = GetContextMock();
+            var agentMock = GetAgentMock();
+            _ = contextMock.SetupSequence(c => c.AgentHandler.Agents)
+                .Returns([agentMock.Object])
+                .Returns([agentMock.Object])
+                .Returns([]);
+
+            _ = contextMock.SetupSequence(c => c.ShouldExit)
+                .Returns(false)
+                .Returns(false)
+                .Returns(true);
```

Comment:
If we return true here, then we would exit even if there were  agents

Reply:
That is not the case, as the agents make it stop after the first iteration. I have however added a comment about it, as I see how that is not directly clear from the code itself.